streaming-binlog/src/main/java/org/openmrs/analytics/DebeziumListener.java

```diff
@@ -0,0 +1,132 @@
+
+        final Predicate isUpdateEvent =
+                header(DebeziumConstants.HEADER_OPERATION).in(
+                        constant(Envelope.Operation.READ.code()),
+                        constant(Envelope.Operation.UPDATE.code()));
+
+        final Predicate isDeleteEvent =
+                header(DebeziumConstants.HEADER_OPERATION).in(
+                        constant(Envelope.Operation.READ.code()),
+                        constant(Envelope.Operation.DELETE.code()));
+
+        /**
+         * Main Change Data Capture (DBZ) entrypoint
+         * @param dbzConfig.
+         */
+        from(getDebeziumConfig())
+                .routeId(DebeziumListener.class.getName() + ".MysqlDatabaseCDC")
+                .log(LoggingLevel.INFO, "Incoming Events: ${body} with headers ${headers}")
+                .choice()
+                    .when(isCreateEvent)
+                        .log(LoggingLevel.TRACE, "CreateEvent Emitted ---> ${body}")
+                        .process(new FhirUriGenerator())
+                        .to(FHIR_ROUTE_HANDLER)
+                        .endChoice()
+                    .when(isUpdateEvent)
+                        .log(LoggingLevel.TRACE, "UpdateEvent Emitted ---> ${body}")
+                        .process(new FhirUriGenerator())
+                        .to(FHIR_ROUTE_HANDLER)
+                        .endChoice()
+                    .when(isDeleteEvent)
+                        .log(LoggingLevel.TRACE, "DeleteEvent Emitted Not Supported ---> ${body}")
+                        .endChoice()
+                    .otherwise()
+                        .log(LoggingLevel.WARN, "Event Not Supported: ${headers[" + DebeziumConstants.HEADER_IDENTIFIER + "]}")
+                .endParent();
+
+        /**
+         * FHIR event worker
+         * Current append mode is suppported i.e any Create or Update will be upserted to the warehouse
+         * append mode should also work for deletion given that openmrs does not delete data instead voids
+         * @param routeDefinition.
+         */
+        from(FHIR_ROUTE_HANDLER)
+                .filter(body().isNotNull())// Filter non-Fhir uri
+                .choice()
+                    .when().simple("${header.fhirResourceUri} == null || ${header.fhirResourceUri} == ''")
+                        .log(LoggingLevel.WARN,"FHIR URL Not Mapped --->  ${headers[" + DebeziumConstants.HEADER_IDENTIFIER + "]}")
+                    .otherwise()
+                        .setBody(simple("${null}")) // Set up body to null
+                        .toD("{{openmrs.serverUrl}}{{openmrs.fhirBaseEndpoint}}${header.fhirResourceUri}" +
+                                "?httpMethod=GET" +
+                                "&authMethod=Basic" +
+                                "&authUsername={{openmrs.username}}" +
+                                "&authPassword={{openmrs.password}}" +
+                                "&authenticationPreemptive=true")
```

Comment:
One more thing I forgot last night: I think you have done some performance evaluations too. You may find [this doc](https://docs.google.com/document/d/1KPiY_9ziEcsh7tAft3QtKuZ7XlDlPxE3yN7xnUO0LJg/edit) useful which summarizes some of my findings on fhir module's performance hot-spots. In particular, you can turn off narrative generation by adding a `_summary` parameter like [here](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/5324bf002f36645c5e5edf20e1cfd7fce3ef39ad/batch/src/main/java/org/openmrs/analytics/FhirSearchUtil.java#L43) which should chop ~70% of fhir module request times.

Reply:
Ooh nice! This will really make the pipeline performant! I'll add this parameter! Thanks @bashir2

Reply:
Yes! this actually makes the pipeline much faster! from ~ 8ms to 3ms :)